phnt/include/ntrtl.h

```diff
@@ -9527,10 +9527,11 @@ RtlComputeCrc32(
  * @sa https://learn.microsoft.com/en-us/previous-versions/bb432254(v=vs.85)
  */
 NTSYSAPI
+_Ret_maybenull_
```

Comment:
It used to be standard to add SAL above the type and after NTSYSAPI statements but a few parsers have been known to ignore anything when the first statement is declspec like NTSYSAPI 

for example the CRT and SDK have moved all the SAL annotations to the first line before anything that's used by a linker in the function signature (i.e. _ACRTIMP _CRTALLOCATOR are linker/declspec):
![image](https://github.com/user-attachments/assets/39aac815-690e-4453-90b5-aaf51e865c4f)

Reply:
`_Ret_maybenull_` also need to include the _check_return_ otherwise it doesn't raise warnings:

```
_Check_return_ 
_Ret_maybenull_
```

Reply:
I believe `_Ret_maybenull_` alone is correct here since the input is allowed to be null. The caller should already know the condition of the input and shouldn't be required to check the condition of the output. And it is meaningless to do so anyway since the pointer will be scrambled.

Reply:
Yes, both of input pointer and output pointer can be `NULL`, so SAL just mark them can be null and no need to check them.